pkg/operator/ceph/csi/spec.go

```diff
@@ -367,6 +367,44 @@ func startDrivers(namespace string, clientset kubernetes.Interface, ver *version
 			return errors.Wrap(err, "failed to load cephfs plugin service template")
 		}
 	}
+
+	pv, _ := clientset.CoreV1().PersistentVolumes().List(metav1.ListOptions{})
+	drivers := []string{CephFSDriverName, RBDDriverName}
+	driversPV := false
+	for _, p := range pv.Items {
+		if p.Spec.CSI == nil {
+			continue
+		}
+		if p.Spec.CSI.VolumeAttributes["clusterID"] == tp.Namespace {
+			for _, d := range drivers {
+				if d == p.Spec.CSI.Driver {
+					driversPV = true
+				}
+			}
+
+		}
+	}
+	if !driversPV {
+		if !EnableRBD {
+			if err := k8sutil.DeleteDaemonset(clientset, tp.Namespace, "csi-rbdplugin"); err != nil {
+				logger.Error(err)
+			}
+			if err := k8sutil.DeleteDeployment(clientset, tp.Namespace, "csi-rbdplugin-provisioner"); err != nil {
+				logger.Error(err)
+			}
+		}
```

Comment:
The drivers also each have a k8s service that needs to be deleted.

Reply:
Do we also need to remove the csi driver registration (from calling `createCSIDriverInfo()`)?

Reply:
Yes we need to delete CSI driverinfo also

Reply:
Done.